spark3/src/main/java/org/apache/iceberg/spark/source/SparkTable.java

```diff
@@ -160,6 +167,43 @@ public WriteBuilder newWriteBuilder(LogicalWriteInfo info) {
     return new SparkWriteBuilder(sparkSession(), icebergTable, info);
   }
 
+  @Override
+  public MergeBuilder newMergeBuilder(LogicalWriteInfo info) {
+    String mode = icebergTable.properties().getOrDefault(WRITE_ROW_LEVEL_MODE, WRITE_ROW_LEVEL_MODE_DEFAULT);
+    ValidationException.check(mode.equals("copy-on-write"), "Unsupported row operations mode: %s", mode);
+    return new SparkMergeBuilder(sparkSession(), icebergTable, info);
+  }
+
+  @Override
+  public boolean canDeleteWhere(Filter[] filters) {
+    if (table().specs().size() > 1) {
+      // cannot guarantee a metadata delete will be successful if we have multiple specs
+      return false;
+    }
+
+    Set<Integer> identitySourceIds = table().spec().identitySourceIds();
+    Schema schema = table().schema();
+
+    for (Filter filter : filters) {
+      // return false if the filter requires rewrite or if we cannot translate the filter
+      if (requiresRewrite(filter, schema, identitySourceIds) || SparkFilters.convert(filter) == null) {
+        return false;
+      }
+    }
+
+    return true;
+  }
+
+  private boolean requiresRewrite(Filter filter, Schema schema, Set<Integer> identitySourceIds) {
+    // TODO: handle dots correctly via v2references
```

Comment:
We may want to add another TODO to detect more cases that don't require rewrites.

Reply:
Will do.

Reply:
Done.